lib/credentials/assume_role_with_web_identity_credentials.js

```diff
@@ -0,0 +1,153 @@
+var AWS = require('../core');
+var fs = require('fs');
+var crypto = require('crypto');
+var STS = require('../../clients/sts');
+
+/**
+ * Represents OIDC credentials from a file on disk
+ * If the credentials expire, the SDK can {refresh} the credentials
+ * from the file.
+ *
+ * ## Using the web identity token file
+ *
+ * This provider is checked by default in the Node.js environment. To use
+ * the provider simply add your OIDC token to a file (ASCII encoding) and
+ * share the filename in either AWS_WEB_IDENTITY_TOKEN_FILE environment
+ * variable or web_identity_token_file shared config variable
+ *
+ * The file contains encoded OIDC token and the characters are
+ * ASCII encoded. OIDC tokens are JSON Web Tokens (JWT).
+ * JWT's are 3 base64 encoded strings joined by the '.' character.
+ *
+ * This class will read filename from AWS_WEB_IDENTITY_TOKEN_FILE
+ * environment variable or web_identity_token_file shared config variable,
+ * and get the OIDC token from filename.
+ * It will also read IAM role to be assumed from AWS_IAM_ROLE_ARN
+ * environment variable or role_arn shared config variable.
+ * This provider gets credetials using the {AWS.STS.assumeRoleWithWebIdentity}
+ * service operation
+ *
+ * @!macro nobrowser
+ */
+AWS.AssumeRoleWithWebIdentityCredentials = AWS.util.inherit(AWS.Credentials, {
+
+  /**
+   * @overload AWS.AssumeRoleWithWebIdentityCredentials()
+   */
+  constructor: function AssumeRoleWithWebIdentityCredentials() {
+    AWS.Credentials.call(this);
+    this.sts = new STS();
```

Comment:
This member can be removed or refer to `WebIdentityCredentials`'s STS client, since this STS client ist not used to make any request.

Reply:
Resolved in https://github.com/aws/aws-sdk-js/pull/2737/commits/d33de140cc6dab88ae3dd72ea31e4978db3a3474